pkg/controller/machine/machine.go

```diff
@@ -801,36 +799,32 @@ func (c *Controller) handleObject(obj interface{}) {
 		glog.V(4).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
 	}
 
-	ownerRef := metav1.GetControllerOf(object)
-	if ownerRef != nil {
-		if ownerRef.Kind != "Machine" {
-			return
-		}
-		machine, err := c.machinesLister.Get(ownerRef.Name)
-		if err != nil {
-			glog.V(4).Infof("ignoring orphaned object '%s' of machine '%s'", object.GetSelfLink(), ownerRef.Name)
-			return
-		}
-
-		glog.V(6).Infof("Processing node: %s (machine=%s)", object.GetName(), machine.Name)
-		c.enqueueMachine(machine)
+	machinesList, err := c.machinesLister.List(labels.Everything())
```

Comment:
Maybe i would like to know if we can move it below https://github.com/kubermatic/machine-controller/pull/309/files#diff-87acff32b00826238809db10ac9ef562R810

Reply:
Actually that is not possible because:

* We check if labels exist
* if no: Iterate machines, enq all that have no nodeRef
* if yes: iterate machines, enq the one that has the ID from the label

Either way, we need the list